shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformChannel.java

```diff
@@ -4,26 +4,536 @@
+        return ActivityInfo.SCREEN_ORIENTATION_FULL_USER;
+      case 0x03: // portraitUp and landscapeLeft
+      case 0x06: // portraitDown and landscapeLeft
+      case 0x07: // portraitUp, portraitDown, and landscapeLeft
+      case 0x09: // portraitUp and landscapeRight
+      case 0x0c: // portraitDown and landscapeRight
+      case 0x0d: // portraitUp, portraitDown, and landscapeRight
+      case 0x0e: // portraitDown, landscapeLeft, and landscapeRight
+        // Android can't describe these cases, so just default to whatever the first
+        // specified value was.
+        switch (firstRequestedOrientation) {
+          case 0x01:
+            return ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
+          case 0x02:
+            return ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE;
+          case 0x04:
+            return ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT;
+          case 0x08:
+            return ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE;
+        }
+    }
+
+    // Execution should never get this far, but if it does then we default
+    // to a portrait orientation.
+    return ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
   }
 
-  public void setMethodCallHandler(@Nullable MethodChannel.MethodCallHandler handler) {
-    channel.setMethodCallHandler(handler);
+  private AppSwitcherDescription decodeAppSwitcherDescription(@NonNull JSONObject encodedDescription) throws JSONException {
+    int color = encodedDescription.getInt("primaryColor");
+    if (color != 0) { // 0 means color isn't set, use system default
+      color = color | 0xFF000000; // color must be opaque if set
+    }
+    String label = encodedDescription.getString("label");
+    return new AppSwitcherDescription(color, label);
   }
 
+  private List<SystemUiOverlay> decodeSystemUiOverlays(@NonNull JSONArray encodedSystemUiOverlay) throws JSONException {
+    List<SystemUiOverlay> overlays = new ArrayList<>();
+    for (int i = 0; i < encodedSystemUiOverlay.length(); ++i) {
+      String encodedOverlay = encodedSystemUiOverlay.getString(i);
+      SystemUiOverlay overlay = SystemUiOverlay.fromValue(encodedOverlay);
+      switch(overlay) {
+        case TOP_OVERLAYS:
+          overlays.add(SystemUiOverlay.TOP_OVERLAYS);
+          break;
+        case BOTTOM_OVERLAYS:
+          overlays.add(SystemUiOverlay.BOTTOM_OVERLAYS);
+          break;
+      }
+    }
+    return overlays;
+  }
+
+  private SystemChromeStyle decodeSystemChromeStyle(@NonNull JSONObject encodedStyle) throws JSONException {
+    Brightness systemNavigationBarIconBrightness = null;
+    // TODO(mattcarroll): add color annotation
+    Integer systemNavigationBarColor = null;
+    // TODO(mattcarroll): add color annotation
+    Integer systemNavigationBarDividerColor = null;
+    Brightness statusBarIconBrightness = null;
+    // TODO(mattcarroll): add color annotation
+    Integer statusBarColor = null;
+
+    if (!encodedStyle.isNull("systemNavigationBarIconBrightness")) {
```

Comment:
subjective nit: if these strings were named `static consts` it would eliminate potential typo errors when using them in multiple places.

Reply:
I think we should probably choose to either do that universally or not at all, so I'll leave this particular code as-is.  We may decide later to move all such messages to constants across all channels.